packages/transaction-controller/src/swaps.ts

```diff
@@ -0,0 +1,370 @@
+
+// Convert to a `type` in a future major version.
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+interface SwapsTokenObject {
+  /**
+   * The name for the network
+   */
+  name: string;
+  /**
+   * An address that the metaswap-api recognizes as the default token
+   */
+  address: string;
+  /**
+   * Number of digits after decimal point
+   */
+  decimals: number;
+}
+
+const ETH_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
+  name: 'Ether',
+  address: DEFAULT_TOKEN_ADDRESS,
+  decimals: 18,
+};
+
+const BNB_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
+  name: 'Binance Coin',
+  address: DEFAULT_TOKEN_ADDRESS,
+  decimals: 18,
+} as const;
+
+const MATIC_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
+  name: 'Matic',
+  address: DEFAULT_TOKEN_ADDRESS,
+  decimals: 18,
+} as const;
+
+const AVAX_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
+  name: 'Avalanche',
+  address: DEFAULT_TOKEN_ADDRESS,
+  decimals: 18,
+} as const;
+
+const TEST_ETH_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
+  name: 'Test Ether',
+  address: DEFAULT_TOKEN_ADDRESS,
+  decimals: 18,
+} as const;
+
+const GOERLI_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
+  name: 'Ether',
+  address: DEFAULT_TOKEN_ADDRESS,
+  decimals: 18,
+} as const;
+
+const ARBITRUM_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
+  ...ETH_SWAPS_TOKEN_OBJECT,
+} as const;
+
+const OPTIMISM_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
+  ...ETH_SWAPS_TOKEN_OBJECT,
+} as const;
+
+const ZKSYNC_ERA_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
+  ...ETH_SWAPS_TOKEN_OBJECT,
+} as const;
+
+export const SWAPS_CHAINID_DEFAULT_TOKEN_MAP = {
+  [CHAIN_IDS.MAINNET]: ETH_SWAPS_TOKEN_OBJECT,
+  [SWAPS_TESTNET_CHAIN_ID]: TEST_ETH_SWAPS_TOKEN_OBJECT,
+  [CHAIN_IDS.BSC]: BNB_SWAPS_TOKEN_OBJECT,
+  [CHAIN_IDS.POLYGON]: MATIC_SWAPS_TOKEN_OBJECT,
+  [CHAIN_IDS.GOERLI]: GOERLI_SWAPS_TOKEN_OBJECT,
+  [CHAIN_IDS.AVALANCHE]: AVAX_SWAPS_TOKEN_OBJECT,
+  [CHAIN_IDS.OPTIMISM]: OPTIMISM_SWAPS_TOKEN_OBJECT,
+  [CHAIN_IDS.ARBITRUM]: ARBITRUM_SWAPS_TOKEN_OBJECT,
+  [CHAIN_IDS.ZKSYNC_ERA]: ZKSYNC_ERA_SWAPS_TOKEN_OBJECT,
+} as const;
+
+export const SWAP_TRANSACTION_TYPES = [
+  TransactionType.swap,
+  TransactionType.swapApproval,
+];
+
+/**
+ * Updates the transaction meta object with the swap information
+ *
+ * @param transactionMeta - The transaction meta object to update
+ * @param transactionType - The type of the transaction
+ * @param swaps - The swaps object
+ * @param swaps.hasApproveTx - Whether the swap has an approval transaction
+ * @param swaps.meta - The swap meta object
+ * @param updateSwapsTransactionRequest - Dependency bag
+ * @param updateSwapsTransactionRequest.isSwapsDisabled - Whether swaps are disabled
+ * @param updateSwapsTransactionRequest.cancelTransaction - Function to cancel a transaction
+ * @param updateSwapsTransactionRequest.controllerHubEmitter - Function to emit an event to the controller hub
+ */
+export async function updateSwapsTransaction(
```

Comment:
In the interest of keeping these changes as small as possible, my understanding is that mobile already supports swaps without the need for this logic and the `transaction-new-swap*` events in the client.

Plus we don't reference this anywhere so ideally we can remove it...

Reply:
We're also not providing the `type` argument on `addTransaction` so none of this logic is used anyway. Removing it would mean we wouldn't need the `TransactionType` also.